File: optimizedProblem72.py
def calc():
    limit = 1000000
    phi = list(range(0, limit + 1))

    out = 0
    for x in range(2, limit):
        if phi[x] == x:
            y = x
            fac = (x - 1) / x
            while y < limit:
                phi[y] = phi[y] * fac
                y += x

        out += phi[x]

    return out

print(calc())

File: OptimizedProblem72.java
public class OptimizedProblem72 {

    public static void main(String[] argv) {
        System.out.println(Problem72.calc());
    }

    public static long calc() {
        int max = 1000000;
        int[] phi = new int[max + 1];
        long out = 0;
        for (int t = 0; t <= max; t++) {
            phi[t] = t;
        }
        for (int i = 2; i <= max; i++) {
            if (phi[i] == i) {
                for (int j = i; j <= max; j += i) {
                    phi[j] =  phi[j] / i * (i - 1);
                }
            }
            out += phi[i];
        }
        return out;
    }

}
